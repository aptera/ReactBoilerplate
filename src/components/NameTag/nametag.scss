.big {
    font-family: sans-serif;
    font-size: 40px;
    font-weight: bold;
}

.yellow {
    color: #F2E394;
}

.orange {
    color: #F2AE72; 
}

img {
    max-width: 100%;
}