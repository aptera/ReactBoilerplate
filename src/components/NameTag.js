import React, {Component} from 'react';
import '../styles/nametag.css';

class NameTag extends Component {

    constructor(props){
        super(props);
        this.state = {
            name: props.name
        };
    }

    render() {
        return (
            <div className="red big">Im {this.state.name}, bitch</div>
        );
    }
}

export default NameTag;