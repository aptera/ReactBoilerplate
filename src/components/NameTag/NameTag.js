import React, {Component} from 'react';
import './nametag.scss';

class NameTag extends Component {

    constructor(props){
        super(props);
        this.state = {
            name: props.name
        };
    }

    render() {
        return (
            <div className={this.props.myClasses}>Im {this.state.name}, bitch!</div>
        );
    }
}

export default NameTag;