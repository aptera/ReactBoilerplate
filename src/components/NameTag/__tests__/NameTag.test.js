import React from 'react';

import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

// import the component to put under test.
import NameTag from '../NameTag';

describe('the nametag component', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <NameTag name="David" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});