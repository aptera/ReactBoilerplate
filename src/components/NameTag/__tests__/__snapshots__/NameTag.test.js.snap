// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the nametag component renders correctly 1`] = `
<div
  className={undefined}
>
  Im 
  David
  , bitch!
</div>
`;
